questionpy_sdk/webserver/static/script.js

```diff
@@ -302,12 +311,66 @@ async function delete_repetition_element(event) {
     }
 
     const response = await post_json('/options/remove-repetition', data);
-    if (response.redirected) {
+
+    if (response.status === 422) {
+        add_errors_to_form(await response.json());
+    } else if (response.redirected) {
         window.location.href = response.url;
     }
 }
 
 
+/**
+ * Adds validation error messages to form inputs.
+ *
+ * @param {Object} errors
+ */
+function add_errors_to_form(errors) {
+    const get_errors_el = (element_el) => Array.from(element_el.children).find(el => el.classList.contains('errors'));
+
+    // Clear error messages
+    const options_form_el = document.getElementById('options_form');
+    if (!options_form_el) {
+        throw new Error(`Could not find options_form!`);
+    }
+    for (const element_el of options_form_el.getElementsByClassName('element')) {
+        const errors_el = get_errors_el(element_el);
+        if (errors_el) {
+            errors_el.hidden = true;
+            errors_el.textContent = '';
+        }
+    }
+
+    // Add error messages
+    for (const [field, message] of Object.entries(errors)) {
+        // Transform form model "parts" to element ID
+        const el_id = field
+            .split('.')
+            // Convert repetition index string to 1-based integers (e.g. "0" -> 1)
+            .map((part) => /^\d+$/.test(part) ? Number.parseInt(part) + 1 : part)
+            .join('_');
+
+        // Try general section...
+        let element_el = document.getElementById(`general_${el_id}`);
+        if (!element_el) {
+            // ...or use first part as section name
+            element_el = document.getElementById(el_id);
+        }
+
+        if (!element_el) {
+            throw new Error(`Could not find form field '${field}'!`);
+        }
+
+        const errors_el = get_errors_el(element_el);
+        if (!errors_el) {
+            throw new Error(`Could not find errors element for field '${field}'!`);
+        }
```

Comment:
Ich denke hier wären wir mit `console.error` und `continue` besser bedient.

Reply:
Ich benutze hier mit voller Absicht `throw new Error`. Es handelt sich um *critical error conditions*. Sollte so ein Fall eintreten, handelt es sich um einen Bug, der im besten Fall reportet wird. Bei Verwendung von `console.error`/`continue` würde das Script ja einfach weiterlaufen und welcher User überwacht schon permanent die dev-Konsole des Browsers und nimmt Fehler dort ernst? Bugs werden dann reportet wenn Dinge offensichtlich und hart failen.

Für eine bessere UX lassen die Exceptions sich außerdem fangen und in einem übersichtlichen User dialog anzeigen, mit der Bitte den Bug zu reporten, unter Anzeige eines Stack trace, Link zum Bug-Tracker, etc. (was nicht Scope dieses PRs ist).

Reply:
Auch ungefangene Errors landen nur in der Konsole. Das Einzige, was dir das `throw` hier im Vergleich zu `console.error` kauft ist, dass zusätzlich zu dem `errors`-Eintrag, der den Fehler verursacht hat, auch alle folgenden `errors`-Einträge verschluckt werden. Das verstehe ich nicht als "offensichtlich und hart failen", sondern ein genauso leises failen auf einer anderweitig weniger nutzbaren Seite.

:+1: für sichtbare Fehlermeldungen im Falle von ungefangenen Errors, das anzugehen macht aber vermutlich erst Sinn nach Umsetzung der von Lars erarbeiteten Zielvorstellung. Diese Diskussion ist daher vermutlich auch müßig, da wir auch den Teil dafür umschreiben werden.

Reply:
> Diese Diskussion ist daher vermutlich auch müßig, da wir auch diesen Teil dafür umschreiben werden.

Genau, ich bin mit beiden ok. Aber merge es jetzt einfach erst mal, da es ja eh bald überarbeitet wird.